sql/src/test/java/org/apache/druid/sql/calcite/CalciteInsertDmlTest.java

```diff
@@ -280,6 +283,242 @@ public void testInsertFromExternal()
         .verify();
   }
 
+  @Test
+  public void testInsertWithPartitionBy()
+  {
+    // Test correctness of the query when only PARTITION BY clause is present
+    RowSignature targetRowSignature = RowSignature.builder()
+                                                  .add("__time", ColumnType.LONG)
+                                                  .add("floor_m1", ColumnType.FLOAT)
+                                                  .add("dim1", ColumnType.STRING)
+                                                  .build();
+
+    Map<String, Object> queryContext = new HashMap<>(DEFAULT_CONTEXT);
+    queryContext.put(QueryContexts.INGESTION_GRANULARITY, "day");
+
+    testInsertQuery()
+        .sql(
+            "INSERT INTO druid.dst SELECT __time, FLOOR(m1) as floor_m1, dim1 FROM foo PARTITION BY 'day'")
```

Comment:
Per earlier; this form of `PARTITION BY` is too implicit; where is `day` coming from? We know it is `__time`, but this makes too many assumptions. Better to use the expression Gian suggested. That way, we can have `PARTITIONED BY FLOOR(__time TO DAY), "cluster", "partition",  "node"`. (Would be good to have a better name for the function; `DAY(__time)`, say, but that's another discussion.)